chain/chunks/src/lib.rs

```diff
@@ -1623,6 +1624,21 @@ impl ShardsManager {
         Ok(())
     }
 
+    pub fn build_block_producer_mapping(
```

Comment:
This is test-only, right? Can we implement this function in a test, instead of it being publicly visible everywhere?

Reply:
Yes, right now it's only used in test code. For me, this piece of code belongs to ShardsManager because it uses only ShardManager data members to compute the result. If you think there is a low probability this method to be reused, I will move it to test code.

Reply:
No super strong opinion here, but I'd define

```rust
fn build_block_producer_mapping(
  shard_manager: &ShardManager,
  prev_block_hash: &CryptoHash,
)
```

in the test code. 

"where the thing is used from" is somewhat more important than "what is needed to implement the thing". public inherent methods should be used for generally re-usable functionality. If you need something in one specific case, better to make it a plain function.